nacos-config-spring-boot-autoconfigure/src/main/java/com/alibaba/boot/nacos/config/util/log/LogAutoFreshProcess.java

```diff
@@ -0,0 +1,139 @@
+import com.alibaba.boot.nacos.config.util.NacosPropertiesBuilder;
+import com.alibaba.nacos.api.common.Constants;
+import com.alibaba.nacos.api.config.ConfigService;
+import com.alibaba.nacos.api.config.listener.AbstractListener;
+import com.alibaba.nacos.api.exception.NacosException;
+import com.alibaba.nacos.client.config.utils.ConcurrentDiskUtil;
+import com.alibaba.nacos.client.config.utils.JvmUtil;
+import com.alibaba.nacos.client.utils.LogUtils;
+import com.alibaba.nacos.common.utils.IoUtils;
+import com.alibaba.nacos.common.utils.StringUtils;
+import com.alibaba.nacos.spring.util.NacosUtils;
+import org.slf4j.Logger;
+import org.springframework.boot.context.logging.LoggingApplicationListener;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.function.Function;
+
+/**
+ * 1. Get the log configuration content through globalproperties
+ * 2. Write the configuration content to the local directory and log it through logging. The config parameter sets the log file path
+ * 3. Register a listener to listen for changes in log configuration. If changes are made, write them to the local directory specified in the previous step
+ *
+ * @author hujun
+ */
+public class LogAutoFreshProcess {
+
+    private static final Logger LOGGER = LogUtils.logger(LogAutoFreshProcess.class);
+
+    private final NacosConfigProperties nacosConfigProperties;
+
+    private final ConfigurableEnvironment environment;
+
+    private static final List<String> LOG_DATA_ID = new ArrayList<>();
+
+    private static final String LOG_CACHE_BASE = System.getProperty("JM.SNAPSHOT.PATH", System.getProperty("user.home")) + File.separator + "nacos"
+            + File.separator + "logConfig";
+
+    static {
+        LOG_DATA_ID.add("logback.xml");
+        LOG_DATA_ID.add("log4j2.xml");
+    }
+
+    public LogAutoFreshProcess(ConfigurableEnvironment environment, NacosConfigProperties nacosConfigProperties) {
+        this.nacosConfigProperties = nacosConfigProperties;
+        this.environment = environment;
+    }
+
+    public void process(Function<Properties, ConfigService> builder) {
+        final String groupName = environment
+                .resolvePlaceholders(nacosConfigProperties.getGroup());
+        for (String dataId : LOG_DATA_ID) {
+            ConfigService configService = builder.apply(buildGlobalNacosProperties());
+            String content = NacosUtils.getContent(configService, dataId, groupName);
+            if (StringUtils.isNotBlank(content)) {
+                writeLogFile(content, dataId);
+                System.setProperty(LoggingApplicationListener.CONFIG_PROPERTY, LOG_CACHE_BASE + File.separator + dataId);
+                registerListener(configService, dataId, groupName);
+            }
+        }
+    }
+
+    public Properties buildGlobalNacosProperties() {
```

Comment:
These properties whether has built in other Configuration class?

Reply:
I'll optimize it